### PR TITLE
Add deployment_id parameter for Azure models

### PR DESCRIPTION
**Pull request by FooQoo**
### Describe the changes you have made:

Fixed an issue where Open Interpreter would fail to chat when an Azure model is specified. The cause was that the Open Interpreter uses the [litellm library](https://github.com/BerriAI/litellm), which has a parameter called `deployment_id`. Setting this parameter allows switching to Azure (see [source code](https://github.com/BerriAI/litellm/blob/58fb6b2b5924041880b9aca1a7f653653a3527a4/litellm/main.py#L593-L595)). However, the current Open Interpreter lacks the logic to set the `deployment_id` parameter, resulting in failed requests to Azure. To resolve this issue, logic has been added to set the `deployment_id` parameter when an Azure model is specified (`model == "azure/{deployment_id}"`).

### Reference any relevant issues (e.g. "Fixes #000"):

No specific issue referenced. This is a general improvement to support Azure models in Open Interpreter.

### Pre-Submission Checklist (optional but appreciated):

- [x] I have included relevant documentation updates (stored in /docs)
- [x] I have read `docs/CONTRIBUTING.md`
- [x] I have read `docs/ROADMAP.md`

### OS Tests (optional but appreciated):

- [ ] Tested on Windows
- [x] Tested on MacOS
- [ ] Tested on Linux

### CODE REVIEW

**Comment by MikeBirdTech** [comment]
Discussion on Discord: https://discord.com/channels/1146610656779440188/1146610657224040451/1210638265254809665

File: interpreter/core/llm/llm.py [hunk 1]
```diff
@@ -166,17 +166,24 @@ def run(self, messages):
...
+            # otherwise, set only the model
+            else:
+                params["model"] = self.model
+
         # Optional inputs
         if self.api_key:
             params["api_key"] = self.api_key
         if self.api_base:
             params["api_base"] = self.api_base
-            params["custom_llm_provider"] = "openai"
```
Comment (MikeBirdTech) [inline interpreter/core/llm/llm.py:179]: What's the benefit of removing this?

Comment (FooQoo) [inline interpreter/core/llm/llm.py:179]: I've opted to remove the line params["custom_llm_provider"] = "openai" from the code. 
This adjustment is based on the understanding that the custom_llm_provider parameter is meant for specifying non-OpenAI language model providers, which isn't necessary when utilizing OpenAI models. 

https://github.com/BerriAI/litellm/blob/58fb6b2b5924041880b9aca1a7f653653a3527a4/litellm/main.py#L440

Comment (FooQoo) [reply interpreter/core/llm/llm.py:179]: Thanks for the comment! I've explained the reason for this in the comments below.

https://github.com/KillianLucas/open-interpreter/pull/1026#discussion_r1501327535

**Comment by FooQoo** [comment]
I tried with the latest version (0.2.3) and was able to chat with no problems using Azure OpenAI!
I'll close this PR.

**Comment by MikeBirdTech** [comment]
Hi @FooQoo 

Happy to hear that you were able to chat using Azure!

Thank you for putting time and effort into this PR and I apologize for letting it slip through the cracks and sitting idle. We have updated how we handle PRs/issues and hope this doesn't happen again. Enjoy 0.2.3!

**Pull request closed without merging**